components/global/analytics.ts

```diff
@@ -11,7 +11,9 @@ export const init = (googleAnalyticsId: string) => {
 }
 
 export const getSessionId = () => {
-  return window.appInsights ? window.appInsights.context.user.id : null
+  return window.appInsights && window.appInsights && window.appInsights.context && window.appInsights.context.user
+    ? window.appInsights.context.user.id
+    : null
 }
```

Comment:
Ah, I wondered about the deep chaining when I saw this first. We only call getSessionId in a single place (Voting.submit) which only serializes the value to JSON. As such, null return value is safe.